Only restrict to image files when onlyImages is active

arknu opened the pull request:
Fixes U4-8034  (http://issues.umbraco.org/issue/U4-8034)

nul800sebastiaan commented on src/Umbraco.Web.UI.Client/src/views/common/overlays/mediaPicker/mediapicker.controller.js:
```diff
@@ -16,7 +16,12 @@ angular.module("umbraco")
+                $scope.acceptedFileTypes = mediaHelper.formatFileTypes(Umbraco.Sys.ServerVariables.umbracoSettings.imageFileTypes);
+            }
+            else {
+                $scope.acceptedFileTypes = "*.*";
```
Should be: 

`mediaHelper.formatFileTypes(Umbraco.Sys.ServerVariables.umbracoSettings.disallowedUploadFiles).replace(/\./g, "!.");`

This is just like other uploaders, allow all files except for the ones in the `disallowedUploadFiles` list. 
I know this is handled server side as well, but might as well do it correctly here too.

arknu replied:
I only discovered the fact that you can do a negative accepted list yesterday, quite useful actually. Was going to update my PR this evening, but you beat me to it ☺ Or should I still push an update this evening?

nul800sebastiaan replied:
Yup, would be great if you have a moment please!

arknu replied:
Will do it this evening then ☺ (in about 3-4 hours)

arknu commented:
@nul800sebastiaan I have updated the PR with your requested changes. I also rebased it to the latest dev-v7 and tested in conjunction with the "Only images" option, now that it works. Everything works as expected, i.e. when "Only images" is checked, you can only upload image files, otherwise you can upload file (as long it is not a disallowed type).

nul800sebastiaan commented:
Awesome, thanks very much!
